src/core/fetch/index.js

```diff
@@ -96,13 +96,32 @@ export function Fetch(Base) {
         // Abort last request
 
         const file = this.router.getFile(path);
-        const req = request(file + qs, true, requestHeaders);
 
         this.isRemoteUrl = isExternal(file);
         // Current page is html
         this.isHTML = /\.html$/g.test(file);
 
         // Load main content
+        const req = Promise.resolve()
```

Comment:
This is perfectly doable. There are two options:
1. Rewrite without promises - might make the code harder to read for modern JS contributors
2. Use a stand-in polyfill, like [promise-polyfill](https://www.npmjs.com/package/promise-polyfill) - in which case the refactor is going to be 0 work, but it will add up to 1kb to the bundle.

WDYT?

Reply:
I'm not a big fan of either option, to be honest. 😉 

If I had to choose one, I'd probably opt to rewrite the code without promises since Docsify includes no other polyfills and we're talking about a total of ~30 lines here. We can worry about modernizing this block and other areas of our codebase if/when we bump to v5 and drop legacy browser support (🤞).

Reply:
Can we just drop legacy support (browsers without Promise are _really really_ old) and bump to V5, without requiring that V5 also had a bunch of feature changes too? We could always bump to V6 for that, but dropping legacy browsers will really help simplify code.